lib/puppet/functions/dns_a.rb

```diff
@@ -1,33 +1,17 @@
 # frozen_string_literal: true
 
-# Retrieves DNS A records and returns it as an array. Each record in the
-# array will be a IPv4 address.
-#
-# An optional lambda can be given to return a default value in case the
-# lookup fails. The lambda will only be called if the lookup failed.
+# [DEPRECATED] Retrieves DNS A records for a domain and returns them as an array.
 Puppet::Functions.create_function(:dns_a) do
+  # @param domain the dns domain to lookup
+  # @param block an optional lambda to return a default value in case the lookup fails
+  # @return [Array[String]] An array of A answers matching domain
```

Comment:
I'd prefer to use [return_type](https://www.puppet.com/docs/puppet/7/functions_ruby_signatures.html#return-type-method) in the dispatch method to declare the return type.

Reply:
done